tieredhashing/tiered_hashing.go

```diff
@@ -418,10 +423,23 @@ func (t *TieredHashing) isCorrectnessPolicyEligible(perf *NodePerf) (float64, bo
 		return result
 	})
 
+	averageSuccess := t.OverAllCorrectnessDigest.Reduce(func(w rolling.Window) float64 {
```

Comment:
is there any way to memo-ize this value / this function?

doing the reduction over 1000 data points on every single request when the general 'overall correctness' ought to be fairly stable  is a lot of duplicated work.

we can try in staging but i am worried this will be a cpu bottleneck when run at scale as it's written